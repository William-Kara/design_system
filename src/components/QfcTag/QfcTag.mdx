import { ArgTypes, Meta } from '@storybook/blocks';
import QfcTag from '@/components/QfcTag/QfcTag.vue';

<Meta title="Elements/Tag" />

# Definition

QfcTags are keywords used to label, classify, and organize items.

## Props

The following props are available for the QfcTag component:

<ArgTypes of={QfcTag} />

# Usage

To use QfcTag, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/bulma/elements/tag.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcTag.scss';

```
