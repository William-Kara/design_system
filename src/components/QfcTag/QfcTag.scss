.qfc-tag.tag {
  height: 1.5rem;
  font-size: 1.12rem;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

