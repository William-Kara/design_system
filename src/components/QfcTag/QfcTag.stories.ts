import Tag from './QfcTag.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof Tag> = {
  component: Tag,
  title: 'Elements/Tag',
  argTypes: {
    label: {
      control: 'text',
    },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    label: 'Ã‰tiquette',
  },
};
