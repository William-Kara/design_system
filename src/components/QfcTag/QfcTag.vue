<template>
  <span class="qfc-tag tag">{{ label }}</span>
</template>

<script lang="ts" setup>
interface Props {
  label: string;
}
defineProps<Props>();
</script>
<style lang="scss" scoped src="./QfcTag.scss" />
