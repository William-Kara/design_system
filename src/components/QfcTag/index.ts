import QfcTag from './QfcTag.vue';

export { QfcTag };
export default QfcTag;
