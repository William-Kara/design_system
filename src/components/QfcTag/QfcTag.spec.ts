import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcTag from './QfcTag.vue';

let wrapper: any = null;

beforeEach(() => {
  wrapper = shallowMount(QfcTag, {
    propsData: {
      label: 'Tag test',
    },
  });
});

describe('QfcTag Component', () => {
  it('mounts properly', () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.text()).toContain('Tag test');
  });

  it('has correct props', () => {
    expect(wrapper.props().label).toBe('Tag test');
  });

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
