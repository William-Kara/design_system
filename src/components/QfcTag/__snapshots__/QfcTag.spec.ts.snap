// Vitest Snapshot v1

exports[`QfcTag Component > renders correctly 1`] = `"<span data-v-91f8913f=\\"\\" class=\\"qfc-tag tag\\">Tag test</span>"`;
